### DISCUSSION TITLE
Google analytics

### DISCUSSION THREAD

**Discussion by aalhommada** [shuding/nextra#1554] [Ideas]
How could I add **google analytics** to Nextra ?
or how can I use [nextjs script](https://nextjs.org/docs/messages/next-script-for-ga) in nextra , is it possible only in _app.tsx instead of Home page  , like the nextjs documentation ?

**Comment by aalhommada** [comment]
I just found this package , and I think it is usufull
[nextjs-google-analytics](https://www.npmjs.com/package/nextjs-google-analytics)

```js
// pages/_app.js
import { GoogleAnalytics } from "nextjs-google-analytics";

const App = ({ Component, pageProps }) => {
  return (
    <>
      <GoogleAnalytics trackPageViews />
      <Component {...pageProps} />
    </>
  );
};

export default App;
```

You can pass your Google Analytics measurement id by setting it on the NEXT_PUBLIC_GA_MEASUREMENT_ID environment variable

```bash
NEXT_PUBLIC_GA_MEASUREMENT_ID="G-XXXXXXXXXX"
```
to your .env.local file

**Reply by mboss37** [reply]
Did this work for you?

**Reply by aalhommada** [reply]
If you mean the package,  No, it is not created by me, I found it useful so I share the info :)

**Reply by mboss37** [reply]
No I mean did it work in terms of getting traffic data in your google analytics account?

**Reply by aalhommada** [reply]
Yes, It is work perfectly.
